conanfile.py

```diff
@@ -54,66 +37,102 @@ class VehicleAppCppSdkConan(ConanFile):
     exports = "version.txt"
 
     # Sources are located in the same place as this recipe, copy them to the recipe
-    exports_sources = ".scripts/common.sh", "build.sh", "install_dependencies.sh", "CMakeLists.txt", "sdk/*", "examples/*", "conanfile.py", ".conan/profiles/*", "version.txt"
+    exports_sources = ".scripts/common.sh", "build.sh", "install_dependencies.sh", "CMakeLists.txt", "sdk/*", "conanfile.py", ".conan/profiles/*", "version.txt", "LICENSE"
+
+    def __run_cmd(self, args):
+        return subprocess.run(args, capture_output=True).stdout.strip().decode("utf-8")
 
     def set_version(self):
         try:
-            git = tools.Git(folder=".")
-            tag = git.get_tag()
-            if tag is not None:
+            tag = self.__run_cmd(["git", "tag", "--points-at", "HEAD"])
+            version = ""
+            if tag:
                 version_tag_pattern = re.compile(r"^v[0-9]+(\.[0-9]+){0,2}")
                 if version_tag_pattern.match(tag):
                     tag = tag[1:] # cut off initial v if a semver tag
-
-            version = tag if tag is not None else git.get_branch()
-            if version == "HEAD (no branch)":
-                version = git.get_commit()
-            self.version = version.replace("/", "_")
+                version = tag
+
+            # if no tag, use branch name or commit hash
+            if not version:
+                version = self.__run_cmd(["git", "symbolic-ref", "-q", "--short", "HEAD"])
+            if not version:
+                version = self.__run_cmd(["git", "rev-parse", "HEAD"])
+            
+            # / is not allowed in conan version
+            self.version = version.replace("/", ".")
             open("./version.txt", mode="w", encoding="utf-8").write(self.version)
-        except:
-            print("Not a git repository, reading version from static file...")
+        except Exception as exc:
+            print(f"Exception catched: {exc}")
+            print("--> Maybe not a git repository, reading version from static file...")
             if os.path.isfile("./version.txt"):
                 self.version = open("./version.txt", encoding="utf-8").read().strip()
             else:
                 raise FileNotFoundError("Missing version.txt!")
-
+        print(f"Determined SDK version: {self.version}")
 
     def config_options(self):
-        if self.settings.os == "Linux":
-            del self.options.fPIC
+        if self.settings.get_safe("os") == "Windows":
+            self.options.rm_safe("fPIC")
 
-    def layout(self):
-        cmake_layout(self, src_folder="sdk")
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def requirements(self):
```

Comment:
should we again mention all requirements explicitly to avoid conflicts?

Reply:
To avoid conflicts I've tried to define the dependencies as minimally as possible now :-)